torch/csrc/jit/interpreter.cpp

```diff
@@ -57,13 +57,73 @@ namespace torch { namespace jit {
 // provide a loop counter
 void desugarTripCounts(Block * b) {
   for(auto n : b->nodes()) {
+
     if(n->kind() == prim::Loop) {
+      auto g = n->owningGraph();
+      auto body_block = n->blocks()[0];
+
+      Value* block_trip_count_input = body_block->addInput();
+      body_block->inputs()[0]->replaceAllUsesWith(block_trip_count_input);
+      body_block->eraseInput(0);
+      n->addOutput();
+
+      Value* max_trip_count_value = n->input(0);
+      // First, replace the max trip count input with the initial trip count.
+      // This means we can treat trip count as a loop-carried dependency and
+      // simply emit an increment in the body of the loop
+      {
+        WithInsertPoint guard(n);
+        Value* initial_trip_count =
+            g->insertNode(g->createConstant(at::zeros(at::CPU(at::kLong), {1})))
+                ->output();
+        // Also move trip count to the end because we need to mutate it after
+        // this transformation and it's easier to append an output in the block
```

Comment:
Rather than work around a missing API in the way we represent code, let's just add the missing `insertInput(0, ...)` API.